geps/gep-1651.md

```diff
@@ -0,0 +1,298 @@
+# GEP-1651: Gateway Routability
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a Gateway so that it is only routable within
+a specific scope (ie. public/private/cluster)
+
+## Goals
+
+- Define a mechanic to set the routability on a Gateway
+- Provide a default set of routability options
+- Provide a way for vendors to support custom options
+
+## Non-Goals
+
+- Per-request/route scope
+- Not a lightweight service mesh
+
+## Introduction
+
+One of the early feature requests for Knative was the ability to deploy an
+application using Knative's HTTP routing support, but make it only available
+within the cluster. I want to be able to specify both the "cluster"
+(service.namespace.svc) and "external" (service.namespace.example.com).
+Gateways using the same GatewayClass on the cluster, but ensure that the
+"cluster" service is only routable within the cluster. This would greatly
+simplify deployment for users over the instructions we have today.
+
+Likewise another use case is to provide load balancing capabilities within a virtual
+private network. Different IaaS providers offer private load balancers to support
+these use cases.
+
+## API
+
+We propose adding a new `routability` field under the `spec.infrastructure` stanza of a Gateway.
+
+### Predefined Routability Values
+
+Implementations MAY implement the following values for 'routability' and MUST abide by
+their defined semantics.
+
+Value | Scope
+-|-
+`Public`|The address is routable on the public internet
+`Private`|The address is routable inside a private network larger than a single cluster (ie. VPC) and MAY include RFC1918 address space
+`Cluster`|The address is routable inside the [cluster's network](https://kubernetes.io/docs/concepts/cluster-administration/networking/#how-to-implement-the-kubernetes-network-model)
+
+Values can be compared semantically - `Public` has a larger scope than `Private`, while `Private` has a larger scope than `Cluster`.
+
+### Vendor prefixed values
+
+Implementations can define custom 'routability' values by specifying a vendor prefix followed
+by a slash `/` and a custom name ie. `com.example.com/my-routability`.
+
+Comparing vendor prefixed scopes with the pre-defined ones in implementation specific.
+
+### Default Routability
+
+The default value of `routability` is implementation specific. It is RECOMMENDED that
+the default `routability` remains consistent for Gateways with the same
+`gatewayClassName`.
```

Comment:
Maybe this would be a reasonable time to consider expanding GatewayClass to include

1) Default Routability
2) Supported Routability Values

Don't need to solve this before merging, but seems like it would be good to at least include as an open question that needs to be resolved before moving to "implementable".

Reply:
Updated the GEP to address this